src/components/List.js

```diff
@@ -51,6 +53,47 @@ const List = ({ firestore }) => {
     setInputText('');
   };
 
+  //* Created two lists
+  const inactiveList = [];
+  const activeList = [];
+
+  //* Used this function to divide active and inactive items
+  filteredList.map(item => {
+    let formatedLastPurchaseDate = dayjs.unix(
+      item.lastPurchasedDate['seconds'],
+    );
+    let formatedToday = dayjs();
+    let difference = formatedToday.diff(formatedLastPurchaseDate, 'd');
+    if (
+      item.numberOfPurchases <= 1 ||
+      parseInt(difference) >= item.nextPurchase * 2
+    ) {
+      inactiveList.push(item);
+    } else {
+      activeList.push(item);
+    }
+  });
+
+  //* Used this function to sort both the lists
+  const lists = arr => {
+    return arr.sort((a, b) => {
+      if (a.nextPurchase === b.nextPurchase) {
+        return a.name.localeCompare(b.name);
+      }
+      return a.nextPurchase > b.nextPurchase ? 1 : -1;
+    });
+  };
+
+  const active = lists(activeList);
+  const inactive = lists(inactiveList);
+
+  //* Added active and inactive status to the items
+  active.map(a => (a['status'] = 'active'));
+  inactive.map(a => (a['status'] = 'inactive'));
+
+  //* Concatenated both the arrays
+  const list = [...active, ...inactive];
```

Comment:
This is a super clever use of the spread operator!!!

I am curious what necessitated using 2 different lists, though?

Reply:
To answer my own question, I'm guessing it would have made sorting more complicated to try to keep it as one list. Is that right? Probably a good call!